### PR TITLE
Collaborators/hai

### PR DESCRIPTION
**Pull request by Zizanzu**
fix search page

### CODE REVIEW

File: home/src/app/pages/register/registerCtrl.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+(function() {
```
Comment (glmanhtu) [inline home/src/app/pages/register/registerCtrl.js]: We have dropped this page, why it still there?

File: home/src/app/pages/goods/goodsCtrl.js [hunk 2]
```diff
@@ -11,6 +11,63 @@
         var vm = this;
         vm.title = 'goodsCtrl';
         vm.getGood = getGood;
+
+  $scope.myInterval = 5000;
```
Comment (glmanhtu) [inline home/src/app/pages/goods/goodsCtrl.js]: What is this ???, we already have slider

File: home/src/app/pages/goods/goods.html [hunk 3]
```diff
@@ -118,4 +118,59 @@
 			</ul>
 		</div>
 	</div>
-</div>	
\ No newline at end of file
+
```
Comment (glmanhtu) [inline home/src/app/pages/goods/goods.html:124]: We don't control transaction, we can't

File: home/src/app/pages/search/searchCtrl.js [hunk 4]
```diff
@@ -46,33 +56,45 @@
...
-            // 10 seconds delay
-            $timeout( function(){
-                vm.setPage(1);
-            }, 1000 );
-            
+            if($rootScope.predicates){
+                console.log('predicates: ');
+                console.log($rootScope.predicates);
+                getSearchPredicate($rootScope.predicates);
+                $timeout( function(){
```
Comment (glmanhtu) [inline home/src/app/pages/search/searchCtrl.js:67]: Why we need settimeout ?

File: home/src/app/services/search/search.service.js [hunk 5]
```diff
@@ -138,7 +144,19 @@
                     'Content-Type': 'application/json'
                 }
             });
+        }
+
+        function prepForBroadcast(msg) {
```
Comment (glmanhtu) [inline home/src/app/services/search/search.service.js:138]: What does this do ?

File: home/src/app/theme/components/searchbar/searchbarCtrl.js [hunk 6]
```diff
@@ -3,9 +3,9 @@
 
     angular.module('ExpertExchange.theme.components')
         .controller('searchbarCtrl', searchbarCtrl);
-    searchbarCtrl.$inject = ['$rootScope', '$scope', '$location', 'toastr', 'searchService', 'DOMAIN_URL', 'recognizeService'];
+    searchbarCtrl.$inject = ['$rootScope', '$scope', '$location', 'toastr', 'searchService', 'DOMAIN_URL', 'recognizeService', '$window'];
     /** @ngInject */
-    function searchbarCtrl($rootScope, $scope, $location, toastr, searchService, DOMAIN_URL, recognizeService) {
+    function searchbarCtrl($rootScope, $scope, $location, toastr, searchService, DOMAIN_URL, recognizeService, $window) {
```
Comment (glmanhtu) [inline home/src/app/theme/components/searchbar/searchbarCtrl.js]: Why you inject $window, I don't see you using it

File: home/src/app/pages/register/registerCtrl.js (hunk 1 above)
Comment (Zizanzu) [reply home/src/app/pages/register/registerCtrl.js]: Ok, I drop it.

File: home/src/app/pages/goods/goodsCtrl.js (hunk 2 above)
Comment (Zizanzu) [reply home/src/app/pages/goods/goodsCtrl.js]: I will delete this function.

File: home/src/app/pages/search/searchCtrl.js (hunk 4 above)
Comment (Zizanzu) [reply home/src/app/pages/search/searchCtrl.js:67]: set timeout to get data from server

File: home/src/app/services/search/search.service.js (hunk 5 above)
Comment (Zizanzu) [reply home/src/app/services/search/search.service.js:138]: Client search good at search page

File: home/src/app/pages/search/searchCtrl.js (hunk 4 above)
Comment (glmanhtu) [reply home/src/app/pages/search/searchCtrl.js:67]: I understand what it is, my question is why we need it ?

File: home/src/app/services/search/search.service.js (hunk 5 above)
Comment (glmanhtu) [reply home/src/app/services/search/search.service.js:138]: Could you expand more details about this ?

File: home/src/app/pages/search/searchCtrl.js (hunk 4 above)
Comment (Zizanzu) [reply home/src/app/pages/search/searchCtrl.js:67]: If we do not set timeout, data would not auto update.

Comment (glmanhtu) [reply home/src/app/pages/search/searchCtrl.js:67]: All other page doesn't use it at all, so, why those still work?

**Pull request closed without merging**